src/embedded_flash/store/mod.rs

```diff
@@ -43,6 +43,28 @@
 //! The data-structure can be configured with the `StoreConfig` trait. By implementing this trait,
 //! the number of possible tags and the association between keys and entries are defined.
 //!
+//! # Properties
+//!
+//! The data-structure provides the following properties:
+//! - When an operation returns success, then the represented multi-set is updated accordingly. For
+//!   example, an inserted entry can be found without alteration until replaced or deleted.
+//! - When an operation returns an error, the resulting multi-set state is described in the error
+//!   documentation.
+//! - When power is lost before an operation returns, the operation will either succeed or be
+//!   rolled-back on the next initialization. So the multi-set would be either left unchanged or
+//!   updated accordingly.
+//!
+//! Those properties rely on the following assumptions:
+//! - Writing a word to flash is atomic. When power is lost, the word is either fully written or not
+//!   written at all.
+//! - Reading a word from flash is deterministic. When power is lost while writing or erasing a word
+//!   (erasing a page containing that word), reading that word repeatedly returns the same result
+//!   (until it is written or its page is erased).
```

Comment:
Side note: this property will stop being true when the flash cell reaches its end-of-life (i.e. beyond max erase/write cycles)

Reply:
Yes indeed. But we can note that the store will reject updates once we reach the flash lifetime, such that it is still possible to read the key content successfully to migrate to a replacement key. However, the user can tweak this behavior by setting up the store with a maximum number of erase cycles that is higher than the one specified by the hardware. It would then be possible to use the store outside the flash lifetime.